apps/designer/app/routes/rest/pages.$projectId.ts

```diff
@@ -65,15 +64,75 @@ const handlePut = async (
   return { ok: true, page: newPage };
 };
 
+const EditPageInput = zfd.formData({
+  id: z.string(),
+  name: z.optional(commonPageInput.name),
+  path: z.optional(commonPageInput.path),
+});
+
+const handlePost = async (
+  projectId: string,
+  request: Request
+): Promise<EditPageData> => {
+  const result = EditPageInput.safeParse(await request.formData());
+  if (result.success === false) {
+    return { errors: result.error.formErrors };
+  }
+  const { id, ...data } = result.data;
+
+  const devBuild = await db.build.loadByProjectId(projectId, "dev");
+
+  if (data.path !== undefined) {
+    const existingPage = utils.pages.findByPath(devBuild.pages, data.path);
+    if (existingPage !== undefined && existingPage.id !== id) {
+      return fieldError("path", `Already used for "${existingPage.name}"`);
+    }
+  }
+
+  await db.build.editPage(devBuild.id, id, data);
+
+  return { ok: true };
+};
+
+const DeletePageInput = zfd.formData({ id: z.string() });
+
+const handleDelete = async (
+  projectId: string,
+  request: Request
+): Promise<DeletePageData> => {
+  const result = DeletePageInput.safeParse(await request.formData());
+  if (result.success === false) {
+    return { errors: result.error.formErrors };
+  }
+  const { id } = result.data;
+
+  const devBuild = await db.build.loadByProjectId(projectId, "dev");
+
+  if (devBuild.pages.homePage.id === id) {
+    return fieldError("id", `Can't delete the home page`);
+  }
+
+  await db.build.deletePage(devBuild.id, id);
+
+  return { ok: true };
```

Comment:
we should normalize this to something we already started using: { status: "ok" } and maybe even wrap it into a function like you did with fieldError and use it everywhere, so that we have this consistently throughout entire application

Reply:
using {status: 'ok'}, gives us a way to also have some other status e.g. status: "terrible" :)